tests/python_package_test/test_engine.py

```diff
@@ -1434,25 +1431,220 @@ def decreasing_metric(preds, train_data):
 
         def constant_metric(preds, train_data):
             return ('constant_metric', 0.0, False)
-
         # test that all metrics are checked (default behaviour)
+        params = {
+            'objective': 'regression',
+            'metric': 'None',
+            'verbose': -1
+        }
         gbm = lgb.train(params, lgb_train, num_boost_round=20, valid_sets=[lgb_eval],
                         feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
                                                          constant_metric(preds, train_data)],
                         early_stopping_rounds=5, verbose_eval=False)
         self.assertEqual(gbm.best_iteration, 1)
-
         # test that only the first metric is checked
+        try:
+            gbm = lgb.train(dict(params, first_metric_only=True), lgb_train,
+                            num_boost_round=20, valid_sets=[lgb_eval],
+                            feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
+                                                             constant_metric(preds, train_data)],
+                            early_stopping_rounds=5, verbose_eval=False)
+            self.assertTrue(False, "LightGBMError should be raised.")
+        except lgb.basic.LightGBMError as e:
+            self.assertEqual(e.args[0], '`first_metric_only` and `feval` are not available at the same time.')
+        # test that various combination of metrics.
+        params = {
+            'boosting_type': 'gbdt',
+            'objective': 'regression',
+            'learning_rate': 0.5,
+            'num_leaves': 5,
+            'metric': 'l2',
+            'verbose': -1,
+            'seed': 123,
+            'bagging_seed': 234,
+            'feature_fraction_seed': 456
```

Comment:
`seed` is enough.
https://lightgbm.readthedocs.io/en/latest/Parameters.html#seed
> this seed is used to generate other seeds, e.g. data_random_seed, feature_fraction_seed, etc.